coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3759,39 +3765,190 @@ def _internal_op_tensor_inplace_fill(context, node):
+
+    # mb.torch_tensor_assign handles multi-dim slicing
+    # so we need to create slice specifications for all other dimensions
+    begin = [0] * x.rank
+    begin[dim] = index
+    begin = mb.concat(values=begin, axis=0)
+    end = x.shape
+    stride = [1] * x.rank
+
+    begin_mask = [True] * x.rank
+    if index.val not in (0, -x.rank):
+        begin_mask[dim] = False
+    end_mask = [True] * x.rank
+    squeeze_mask = [False] * x.rank
+    squeeze_mask[dim] = True
+
+    updated_x = _translate_torch_tensor_assign(
+        x=x,
+        updates=updates,
+        begin=begin,
+        end=end,
+        stride=stride,
+        begin_mask=begin_mask,
+        end_mask=end_mask,
+        squeeze_mask=squeeze_mask,
+        name=node.name,
+    )
+    context.add(updated_x)
+
+
+@register_torch_op
+def slice_scatter(context, node):
+    inputs = _get_inputs(context, node, min_expected=2)
+    x = inputs[0]
+    updates = inputs[1]
+    dim = 0 if len(inputs) <= 2 else inputs[2].val
+    start = 0 if len(inputs) <= 3 else inputs[3]
+    end = x.shape[dim] if len(inputs) <= 4 else mb.minimum(x=inputs[4], y=x.shape[dim])
+    step = 1 if len(inputs) <= 5 else inputs[5]
+
+    assert dim is not None, "slice dim must be known at compile time"
+    assert 0 <= dim and dim < x.rank
+
+    # mb.torch_tensor_assign handles multi-dim slicing
+    # so we need to pad start, end, step from scalar to x.rank
+    starts = [0] * x.rank
+    starts[dim] = start
+    starts = mb.concat(values=starts, axis=0)
+    ends = list(x.shape)
+    ends[dim] = end
+    ends = mb.concat(values=ends, axis=0)
+    steps = [1] * x.rank
+    steps[dim] = step
+    steps = mb.concat(values=steps, axis=0)
+
+    # mb.torch_tensor_assign also have masks
+    begin_mask = [True] * x.rank
+    if start.val not in (0, -x.rank):
+        begin_mask[dim] = False
+    end_mask = [True] * x.rank
+    if end.val is None or end.val < x.shape[dim]:
+        end_mask[dim] = False
+    squeeze_mask = [False] * x.rank
+
+    updated_x = _translate_torch_tensor_assign(
+        x=x,
+        updates=updates,
+        begin=starts,
+        end=ends,
+        stride=steps,
+        begin_mask=begin_mask,
+        end_mask=end_mask,
+        squeeze_mask=squeeze_mask,
+        name=node.name,
+    )
+    context.add(updated_x)
+
+
+@register_torch_op
+def index_put(context, node):
+    inputs = _get_inputs(context, node, min_expected=3)
+    x = inputs[0]
     indices = inputs[1]
     values = inputs[2]
-    accumulate = inputs[3].val
-    rank = x.rank
+    accumulate = False if len(inputs) < 4 else inputs[3].val
     mode = "add" if accumulate else "update"
 
-    indices_type = indices[0].sym_type.get_primitive()
+    assert isinstance(indices, list), "indices must be a list of tensors"
+    # Usually indices is a list of non-None tensors, so we stack them and feed to mb.scatter_nd
+    # However, when there exists a whole slice, that index is represented as None
+    exist_slice = False
```

Comment:
I think it's better to do something like this:
`exist_slice = any(map(lambda i: i is None, indices))`

Reply:
Wow this is awesome! Much more concise! Revised as suggested